tiledb/ml/models/tensorflow_keras.py

```diff
@@ -200,102 +242,128 @@ def load(
-                layer_names = []
-                weight_names = []
-                weight_values = []
-                for layer in sorted(self.artifact.layers, key=attrgetter("name")):
-                    weights = layer.trainable_weights + layer.non_trainable_weights
-                    weight_values.append(
-                        pickle.dumps(tf.keras.backend.batch_get_value(weights))
-                    )
-                    weight_names.append(
-                        pickle.dumps([w.name.encode("utf8") for w in weights])
-                    )
-                    layer_names.append(layer.name)
-
-                tf_model_tiledb[:] = {
-                    "layer_name": np.array(layer_names),
-                    "weight_values": np.array(weight_values),
-                    "weight_names": np.array(weight_names),
-                    # TODO (TeamML) Fix array scheme to avoid optimizer_weight repetition. Nullable
-                    "optimizer_weights": np.array(
-                        [
-                            serialized_optimizer_weights
-                            for _ in range(len(self.artifact.layers))
-                        ]
-                    ),
-                }
-
-            # Insert all model metadata
-            model_metadata = saving_utils.model_metadata(
-                self.artifact, include_optimizer
-            )
-            for key, value in model_metadata.items():
-                tf_model_tiledb.meta[key] = json.dumps(
-                    value, default=get_json_type
-                ).encode("utf8")
+    def get_weights(self, timestamp: Optional[Timestamp] = None) -> List[np.ndarray]:
+        """
+        Returns the weights of a Tensorflow Keras model.
+        """
+        with tiledb.open(self.uri, ctx=self.ctx, timestamp=timestamp) as model_array:
+            model_meta = dict(model_array.meta.items())
 
-            self.update_model_metadata(array=tf_model_tiledb, meta=meta)
+            try:
+                model_weights_size = model_meta["model_size"]
+            except KeyError:
+                raise Exception(
+                    f"model_weights_size metadata entry not present in {self.uri}"
+                    f" (existing keys: {set(model_meta)})"
+                )
 
-    def _serialize_optimizer_weights(self, include_optimizer: bool = True) -> bytes:
-        """Serialize optimizer weights"""
+            md_weights_contents: np.ndarray = model_array[0:model_weights_size]["model"]
+            model_weights = pickle.loads(md_weights_contents.tobytes())
+
+        return model_weights  # type: ignore
+
+    def get_optimizer_weights(
+        self, timestamp: Optional[Timestamp] = None
+    ) -> List[np.ndarray]:
+        """
+        Returns the weights of the optimizer of a Tensorflow Keras model.
+        """
+        with tiledb.open(self.uri, ctx=self.ctx, timestamp=timestamp) as model_array:
+            model_meta = dict(model_array.meta.items())
```

Comment:
No need to copy model_array.meta to a new dict, it supports the necessary dict methods